core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1048,7 +1048,9 @@ class UnifiedLog(@volatile var logStartOffset: Long,
           if (duplicateBatch.isPresent) {
             return (updatedProducers, completedTxns.toList, Some(duplicateBatch.get()))
           }
+        }
 
+        if (origin == AppendOrigin.CLIENT || origin == AppendOrigin.COORDINATOR) {
           // Verify that if the record is transactional & the append origin is client, that we either have an ongoing transaction or verified transaction state.
```

Comment:
Comment should probably be updated.

Reply:
Good point. I will also update when David merges his change since it subtly affects this one.